icu4c/source/common/uloc.cpp

```diff
@@ -1103,6 +1103,14 @@ ulocimp_setKeywordValue(const char* keywords,
  */
 #define _isTerminator(a)  ((a==0)||(a=='.')||(a=='@'))
 
+inline bool _isBCP47Extension(const char* p) {
+    return p[0] == '-' &&
```

Comment:
Should this be lenient with `-` vs `_`?
I *think* we usually are... but may be wrong.